### PR TITLE
Feature mdr 49 contact us

### PR DESCRIPTION
**Pull request by labaynecss**

### CODE REVIEW

**Review by acostales08 — APPROVED** [review]

File: src/components/Content/contactUs/ContactForm.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  firstName: requiredString("Your firstname is required."),
+  lastName: requiredString("Your lastname is required."),
+  email: requiredString("Your email is required.").email(),
+  message: requiredString("Message is required"),
+  pos: z.boolean(),
+  tms: z.boolean(),
+});
+
+const ContactForm = () => {
+  const methods = useForm({
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:21]: change from "methods" to "form"

File: src/components/Content/contactUs/ContactForm.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const onSubmit = (data: any) => {
+    console.log(data);
+    // Additional steps such as sending the form to a server or processing the data
+  };
+
+  return (
+    <Container style={{ marginTop: '0' }}>
+      <FormProvider {...methods}>
+        <Grid container spacing={2}>
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:34]: Use our built in component "ControlledGrid"

File: src/components/Content/contactUs/ContactForm.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          <Grid item xs={12}>
+            <ControlledTextField
+              control={methods.control}
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:37]: do not use "method.control" instead use "control" only

File: src/components/Content/contactUs/ContactForm.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+const schema = z.object({
+  firstName: requiredString("Your firstname is required."),
+  lastName: requiredString("Your lastname is required."),
+  email: requiredString("Your email is required.").email(),
+  message: requiredString("Message is required"),
+  pos: z.boolean(),
+  tms: z.boolean(),
+});
+
+const ContactForm = () => {
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:20]: Your code is too straight forward. Isolate your grid and textfield to different function example.

Also you should consider using 

- useFormContext inside your ContactUsForm

const ContactUsForm = () => {
//...here you can put the Grid and ControlledTextField
}


export const ContactUs = () => {
const form = useForm({})
return (
<FormProvider {...form}></FormProvider>
)
}

File: src/components/Content/contactUs/ContactForm.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+          <Grid item xs={12}>
+            <ControlledTextField
+              control={methods.control}
+              name="firstName"
+              label="Firstname"
+            />
+          </Grid>
+          <Grid item xs={12} >
+            <ControlledTextField
+              control={methods.control}
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:44]: Also when implementing ControlledTextField , there must be required , shouldUnregister

File: src/components/Content/contactUs/ContactForm.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            style={{
+              marginTop: '25px',
+              marginLeft: '15px'
+            }}
+                    
+            />
+          <Grid item xs={12}>
+            <FormControlLabel
+              control={
+                <Checkbox
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:68]: we have a component for checkbox named "ControlledCheckbox"

File: src/components/Content/contactUs/ContactForm.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                />
+              }
+              label="POS"
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  {...methods.register('tms')}
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:80]: we shouldn't use ...methods.register. we are just using control={control}

File: src/components/Content/contactUs/ContactForm.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+              label="Message"
+              multiline
+              rows={4}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <NormalButton
+              variant="outlined"
+              color="primary"
+              onClick={methods.handleSubmit(onSubmit)}
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactForm.tsx:100]: from "methods.handleSubmit" must be onClick={handleContinue}

const handleContinue = () => {
handleSubmit((values) => {}) return false
}

File: src/components/Content/contactUs/ContactUsContent.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+
+type ContactUsProps = {
+  children: React.ReactNode;
+};
+const ContactUs: React.FC<ContactUsProps> = ({ children }) => {
```
Comment (JMSevilla) [inline src/components/Content/contactUs/ContactUsContent.tsx:7]: Don't create ContactUsProps and implement here. 

Instead use "React.PropsWithChildren<{}>"

File: src/pages/contactUs.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+              y2="474.645"
+              gradientUnits="userSpaceOnUse">
+              <stop stopColor="#9089FC" />
+              <stop offset={1} stopColor="#FF80B5" />
+            </linearGradient>
+          </defs>
+        </svg>
+      </div>
+      <ContactUs
+        children={
```
Comment (JMSevilla) [inline src/pages/contactUs.tsx:33]: We are now preventing this code we are just not yet doing some migration to this. 

If your component is expecting a children react node. then use 

<ContactUs>
//...your design here
</ContactUs>

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by acostales08 — COMMENTED** [review]
wrong file

**Pull request closed without merging**